src/_pytest/pytester.py

```diff
@@ -879,16 +881,19 @@ class reprec:
             sys.stderr.write(err)
 
         res = RunResult(reprec.ret, out.split("\n"), err.split("\n"), time.time() - now)
-        res.reprec = reprec
+        res.reprec = reprec  # type: ignore
         return res
 
-    def runpytest(self, *args, **kwargs):
+    def runpytest(self, *args, **kwargs) -> RunResult:
         """Run pytest inline or in a subprocess, depending on the command line
         option "--runpytest" and return a :py:class:`RunResult`.
 
         """
         args = self._ensure_basetemp(args)
-        return self._runpytest_method(*args, **kwargs)
+        if self._method == "inprocess":
```

Comment:
I find it more symmetric to write as

```
if self._method == "inprocess": ...
elif self._method == "subprocess": ...
else:
     assert False, "Unrecognized runpytest option: {}".format(self._method)
```

Also, if we sometime decide to change this `_method` variable to be a `Literal["inprocess", "subprocess"]`, then we can add exhaustiveness checking in the `else` branch, so mypy warns if we forget to handle a case.

Reply:
> Literal["inprocess", "subprocess"]

Cool, could do this here already then.

Reply:
But of course it would not prevent runtime error really, for that we should maybe check it in `__init__` then already?

Reply:
Changed it to RuntimeError and your style.
It is using "choices" however already: https://github.com/blueyed/pytest/blob/bf4c45df96346684330943cd9db71af013a2b93a/src/_pytest/pytester.py#L46